modules/nw-how-nw-iface-selected.adoc

```diff
@@ -0,0 +1,78 @@
+
+After the script determines the correct NIC, the script creates the `/etc/systemd/system/kubelet.service.d/20-nodenet.conf` file.
+The `20-nodenet.conf` file sets the `KUBELET_NODE_IP` environment variable to the IP address that was selected.
+
+When the kubelet service starts, it reads the value of the environment variable from the `20-nodenet.conf` file and sets the IP address as the value to the `--node-ip` kubelet command-line argument.
+As a result, the kubelet service uses the selected IP address as the node IP address.
+
+If hardware or networking is reconfigured after installation, it is possible that the script for the `nodeip-configuration.service` unit can select a different NIC after a reboot and prevent {product-title} communication.
+
+If {product-title} communication is disrupted because a different NIC is selected, one strategy for overriding the selection process is to set the correct IP address explicitly.
+The following list identifies the high-level steps and considerations:
+
+* Create a shell script that determines the IP address to use for {product-title} communication. Have the script create a configuration file such as `/etc/systemd/system/kubelet.service.d/98-nodenet-override.conf` and set the `KUBELET_NODE_IP` environment variable to the IP address.
+
+* Create a machine config object with the shell script as a base64-encoded string and use the Machine Config Operator to deploy the script to the nodes at a file system path such as `/usr/local/bin/override-node-ip.sh`.
+
+* Do not overwrite the `/etc/systemd/system/kubelet.service.d/20-nodenet.conf` file. Specify a file name with a lexicographically higher value such as `98-nodenet-override.conf` in the same directory path. The goal is to have the custom systemd unit file run after `20-nodenet.conf` and override the value of the environment variable.
+
+* Ensure that `systemctl daemon-reload` runs after the systemd unit file runs. The simplest method is to specify `ExecStart=systemctl daemon-reload` in the machine config.
+
+.Sample machine config to override the network interface for kubelet
+[source,yaml,subs="attributes+"]
+----
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+     machineconfiguration.openshift.io/role: worker
+  name: 98-nodenet-override
+spec:
+  config:
+    ignition:
+      version: {ign-config-version}
+    storage:
+      files:
+      - contents:
+          source: data:text/plain;charset=utf-8;base64,<encoded_script>
+        mode: 0755
+        overwrite: true
+        path: /usr/local/bin/override-node-ip.sh
+    systemd:
+      units:
+      - contents: |
+          [Unit]
+          Description=Override node IP detection
+          Wants=network-online.target
+          Before=kubelet.service
+          After=network-online.target
+          [Service]
+          Type=oneshot
+          ExecStart=/usr/local/bin/override-node-ip.sh
```

Comment:
can we detail the content of file /usr/local/bin/override-node-ip.sh , like https://bugzilla.redhat.com/show_bug.cgi?id=1920282#c16

Reply:
My first instinct is that we can't support customers one-off scripts and that if we had one script that always worked, we'd have baked that logic into the nodeip-configuration.service software.

I'm sympathetic to our customers' predicament, especially for customers who have a consistent interface name because that might be particularly easy to script. Is there a tactic in the script that we should share in a general fashion?

Reply:
I think we can just list one example here use steps (https://bugzilla.redhat.com/show_bug.cgi?id=1920282#c16 or https://bugzilla.redhat.com/show_bug.cgi?id=1920282#c18) and add comment to notice customer to update this file `/usr/local/bin/override-node-ip.sh` according to their owned cluster.

Reply:
@zhaozhanqi , I respect your commitment to our customers, but I am not comfortable including a script in the product documentation--even with an indication that the script is not applicable in all cases.

Reply:
yes,  the content of shell script may be different for different cluster. However the final propose of the shell script is generate the config file `98-nodenet-override.conf`. So I think we can give an example to make it more easy to understand.  just IMO.